storage/src/mmr/benches/prove_many_elements.rs

```diff
@@ -0,0 +1,76 @@
+use commonware_cryptography::{Digest, Hasher, Sha256};
+use commonware_storage::mmr::mem::Mmr;
+use criterion::{criterion_group, Criterion};
+use rand::{rngs::StdRng, seq::SliceRandom, RngCore, SeedableRng};
+
+const SAMPLE_SIZE: usize = 100;
+
+fn bench_prove_many_elements(c: &mut Criterion) {
+    for n in [10_000, 100_000, 1_000_000, 5_000_000, 10_000_000] {
+        // Populate MMR
+        let mut mmr = Mmr::<Sha256>::new();
+        let mut positions = Vec::with_capacity(n);
+        let mut elements = Vec::with_capacity(n);
+        let mut sampler = StdRng::seed_from_u64(0);
+        for i in 0..n {
+            let mut digest = vec![0u8; Sha256::len()];
+            sampler.fill_bytes(&mut digest);
+            let element = Digest::from(digest);
+            let pos = mmr.add(&element);
+            positions.push((i, pos));
+            elements.push(element);
+        }
+        let root_hash = mmr.root_hash();
+
+        // Generate SAMPLE_SIZE random starts without replacement
+        for range in [2, 5, 10, 25, 50, 100] {
```

Comment:
Not entirely sure what we'd expect to be proving in practice, but I'd guess during syncing we'd be proving ranges quite a bit larger than 100?